src/index.ts

```diff
@@ -23,13 +23,20 @@ const expression = /^(19|20)?(\d{6}\d{4}|(?!19|20)\d{10})$/;
  */
 function ssnIsValid(value) {
   let dateArr, dateStr;
+  let isSsnDateValid: boolean;
+  isSsnDateValid = false;
 
+  // Replace all non-word characters with '-', e.g. "1900-01-01:1234" will result in "1900-01-01-1234"
   value = value.replace(/\W/g, '-');
+  
+  // If the SSN is in the format 19000101-1234 (or 000101-1234) this will result in an array with two entries
```

Comment:
what if the input is

> 199802111234

or 

> 9802111234

Reply:
That is actually not a problem because it then does not get split and gets handled by "else" part from the original logic.
In the case of a Swedish SSN in the format 19000101-1234 later just the '-' is removed and also the original logic applied to convert it to a string that (if  it is a valid SSN) can be parse it into a date.